src/components/Features/TwoColLayout.js

```diff
@@ -16,17 +16,63 @@ import OrchestrationLight from "./images/orchestration-light.svg";
 import OrchestrationDark from "./images/orchestration.svg";
 import { useStyledDarkMode } from "../../theme/app/useStyledDarkMode.js";
 const TwoColLayout = () => {
-  const containerRef = useRef(null);
+  const containerRefs = useRef([]);
+  const contentRefs = useRef([]);
   const { isDark } = useStyledDarkMode();
+
+  useEffect(() => {
+    const options = {
+      threshold: 0.1,
+    };
+
+    const observer = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          entry.target.classList.add("fade-in");
+        } else {
+          entry.target.classList.remove("fade-in");
+        }
+      });
+    }, options);
+
+    containerRefs.current.forEach((ref, index) => {
+      if (ref) {
+        observer.observe(ref);
+        // Adding classes for slide-in animations
+        ref.classList.add(index % 2 === 0 ? "left-slide-in" : "right-slide-in");
+      }
+    });
+
+    contentRefs.current.forEach((ref) => {
+      if (ref) {
+        observer.observe(ref);
+      }
+    });
+
+    return () => {
+      containerRefs.current.forEach((ref) => {
+        if (ref) {
+          observer.unobserve(ref);
+        }
+      });
+
+      contentRefs.current.forEach((ref) => {
```

Comment:
@hargunkaur286 is this repetitive conditions can be reused or minimized?

Reply:
@hargunkaur286 @sudhanshutech are there any additional thoughts here?